lib/Twig/Environment.php

```diff
@@ -803,18 +814,28 @@ public function getExtension($name)
      */
     public function addExtension(Twig_ExtensionInterface $extension)
     {
-        $name = $extension->getName();
+        $name = get_class($extension);
+
+        $m = new ReflectionMethod($extension, 'getName');
+        $legacyName = null;
+        // we don't trigger the deprecation notice for the internal extensions
+        // as people might have extended them (even if this is deprecated and not recommended anymore)
+        if ('Twig_Extension' !== $m->getDeclaringClass()->getName() && 'Twig_Extension_' !== substr($name, 0, 15) && 'Mock_Twig_Extension' !== substr($name, 0, 19)) {
+            @trigger_error(sprintf('Defining the getName() method in the "%s" extension is deprecated since 1.26 and will be removed in 2.0. It\'s not used anymore internally.', $name), E_USER_DEPRECATED);
```

Comment:
should we really trigger a deprecation when they define the method ? Defining an unused method is harmless, and allows to keep compat with older versions